cmd/sink/filters/filters.go

```diff
@@ -128,36 +127,61 @@ func (f *Filters) createFilters(
 	if err != nil {
 		return archiveMap, deleteMap, archiveOnDeleteMap, err
 	}
+
 	var errList []error
 	for _, resource := range resources {
 		gvk := schema.FromAPIVersionAndKind(resource.Selector.APIVersion, resource.Selector.Kind)
 		namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, gvk)
-		if resource.ArchiveWhen != "" {
-			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk.String()])
-			if err != nil {
-				errList = append(errList, err)
-			} else {
-				archiveMap[namespaceGvk] = *archiveExpr
-			}
+
+		errList = addLocalFilters(resource.ArchiveWhen, globalArchive[gvk], archiveMap, namespaceGvk, errList)
+		errList = addLocalFilters(resource.DeleteWhen, globalDelete[gvk], deleteMap, namespaceGvk, errList)
+		errList = addLocalFilters(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk], archiveOnDeleteMap, namespaceGvk, errList)
```

Comment:
I think errors from calls to `addLocalFilters` will be lost. I think there needs to be a call to append
```suggestion
		errList = append(errList, ...addLocalFilters(resource.ArchiveWhen, globalArchive[gvk], archiveMap, namespaceGvk, errList))
		errList = append(errlist, ...addLocalFilters(resource.DeleteWhen, globalDelete[gvk], deleteMap, namespaceGvk, errList))
		errList = append(errlist, ...addLocalFilters(resource.ArchiveOnDelete, globalArchiveOnDelete[gvk], archiveOnDeleteMap, namespaceGvk, errList))
```

Reply:
The errors are now appended to `errList` within the function itself. I can change it back to append them in the caller function if that makes it easier.

Reply:
> to have the errors get logged by the sink, appending needs to happen in both places.

Why? The error list is passed into the method. New errors are appended to the list and returned, and that return value become the error list.  What is wrong?

Reply:
> The error list is passed into the method

I missed that. This request is not relevant